html/includes/forms/get-host-dependencies.inc.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * LibreNMS
+ *
+ * Copyright (c) 2017 Aldemir Akpinar <https://github.com/aldemira/>
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ */
+ 
+if (is_admin() === false) {
+    $status = array('status' => 1, 'message' => 'You need to be admin');
+} else {
+    if (isset($_POST['viewtype'])) {
+        if ($_POST['viewtype'] == 'fulllist') {
+            $count_query = "SELECT count(device_id) from devices";
+
+            $deps_query = "SELECT  a.device_id as id, a.hostname as hostname, GROUP_CONCAT(b.hostname) as parents, GROUP_CONCAT(b.device_id) as parentid FROM devices as a LEFT JOIN device_relationships a1 ON a.device_id=a1.child_device_id LEFT JOIN devices b ON b.device_id=a1.parent_device_id GROUP BY a.device_id, a.hostname";
+
+            if (isset($_POST['format'])) {
+                if (isset($_POST['searchPhrase']) && !empty($_POST['searchPhrase'])) {
+                    #This is a bit ugly
+                    $deps_query = "SELECT * FROM (".$deps_query;
+                    $deps_query .= " ) as t WHERE t.hostname LIKE ? OR t.parents LIKE ? ";
+                    $deps_query .= " ORDER BY t.hostname";
+                } else {
+                    $deps_query .= " ORDER BY a.hostname";
+                }
+
+                if (is_numeric($_POST['rowCount']) && is_numeric($_POST['current'])) {
+                    $rows = $_POST['rowCount'];
+                    $current = $_POST['current'];
+                    $deps_query .= " LIMIT ".$rows * ($current - 1).", ".$rows;
+                }
+            } else {
+                $deps_query .= " ORDER BY a.hostname";
+            }
+
+            if (isset($_POST['format']) && !empty($_POST['searchPhrase'])) {
+                $searchphrase = '%'.$_POST['searchPhrase'].'%';
```

Comment:
You should call mres() on any user input like `$_POST['searchPhrase']`

Reply:
Just use `$vars[$name]` instead of `$_POST[$name]` or `$_GET[$name]` as that runs it through `clean()` which does `mres()`.